sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -2390,3 +2395,41 @@ def _weights(dist):
 
     with pytest.raises(ValueError, match=msg):
         est.predict_proba([[1.1, 1.1]])
+
+
+def test_neighbor_classifiers_loocv():
+    """Check that `predict` and related functions work fine with X=None"""
+    X, y = datasets.make_blobs(n_samples=500, centers=5, n_features=2, random_state=0)
+
+    models = [
+        neighbors.KNeighborsClassifier(n_neighbors=10),
+        neighbors.RadiusNeighborsClassifier(radius=5.0),
+    ]
+
+    for knn in models:
+        loocv = cross_val_score(knn, X, y, cv=LeaveOneOut())
+
+        knn.fit(X, y)
+
+        assert np.all(loocv == (knn.predict(None) == y))
+        assert np.mean(loocv) == knn.score(None, y)
+        assert knn.score(None, y) < knn.score(X, y)
```

Comment:
Could you parametrize this and the test below with the models in the following manner? This will allow us to remove the inner loop.
```suggestion
@pytest.mark.parametrize("nn_model", [
    neighbors.KNeighborsRegressor(n_neighbors=10),
    neighbors.RadiusNeighborsRegressor(radius=0.5),
])
def test_neighbor_classifiers_loocv(nn_model):
    """Check that `predict` and related functions work fine with X=None"""
    X, y = datasets.make_blobs(n_samples=500, centers=5, n_features=2, random_state=0)
    loocv = cross_val_score(nn_model, X, y, cv=LeaveOneOut())
    nn_model.fit(X, y)
    assert np.all(loocv == (nn_model.predict(None) == y))
    assert np.mean(loocv) == nn_model.score(None, y)
    assert knn.score(None, y) < nn_model.score(X, y)
```

Reply:
Sure, if it's the preferred way, then let's do it like that. Done and pushed.